search_service/proxy/atlas.py

```diff
@@ -0,0 +1,230 @@
+            #     col_attrs = col_entity['attributes']
+            #     columns.append(col_attrs.get(self.NAME_KEY))
+
+            table = Table(name=attrs.get(self.NAME_ATTRIBUTE),
+                          key=attrs.get(self.NAME_ATTRIBUTE),
+                          description=attrs['description'],
+                          cluster=db_cluster,
+                          database=db_name,
+                          schema_name=db_name,
+                          column_names=columns,
+                          tags=tags,
+                          last_updated_epoch=attrs.get('updateTime'))
+
+            table_results.append(table)
+
+        return table_results
+
+    @timer_with_counter
+    def fetch_search_results_with_field(self, *,
+                                        query_term: str,
+                                        field_name: str,
+                                        field_value: str,
+                                        page_index: int = 0) -> SearchResult:
+        """
+        Query Atlas and return results as list of Table objects.
+        Per field name we have a count query and a query for the tables.
+        https://atlas.apache.org/Search-Advanced.html
+
+        :param query_term: search query term
+        :param field_name: field name to do the searching(e.g schema_name, tag_names)
+        :param field_value: value for the field for filtering
+        :param page_index: index of search page user is currently on
+        :return: SearchResult Object
+        :return:
+        """
+
+        sql = f"Table from Table where false"
+        count_sql = f"{sql} select count()"
+        if field_name == 'tag':
+            sql = f"from Table where Table is '{field_value}'"
+            count_sql = f"{sql} select count()"
+        elif field_name == 'schema':
+            sql = f"from Table where db.name like '{field_value}'"
+            count_sql = f"{sql} select count()"
+        elif field_name == 'table':
+            sql = f"from Table where name like '{field_value}'"
+            count_sql = f"{sql} select count()"
+        elif field_name == 'column':
+            sql = f"hive_column where name like '{field_value}' select table"
+            # TODO nanne: count tables instead of columns
+            count_sql = f"hive_column where name like '{field_value}' select count()"
+
+        LOGGING.debug(f"Used following sql query: {sql}")
+        tables: List[Table] = []
+        count_value = 0
+        try:
+            # count results
+            count_params = {'query': count_sql}
+            count_results = list(self.atlas.search_dsl(**count_params))[0]
+            count_value = count_results._data['attributes']['values'][0][0]
+
+            params = {'query': f"{sql} limit {self.page_size} offset {page_index * self.page_size}"}
+            search_results = self.atlas.search_dsl(**params)
+            if count_value > 0 and page_index * self.page_size <= count_value:
```

Comment:
nit: Idiomatic python suggests us to use `truthy` and `falsy`. 

`if count_value > 0` -> `if count_value`

Reply:
this was not resolved btw